R/gg-plots.R

```diff
@@ -242,74 +256,128 @@ ggally_density <- function(data, mapping, ...){
 ggally_cor <- function(
   data,
   mapping,
-  alignPercent = 0.6,
-  method = "pearson", use = "complete.obs",
-  corAlignPercent = NULL, corMethod = NULL, corUse = NULL,
-  displayGrid = TRUE,
-  ...
-){
+  ...,
+  stars = TRUE,
+  method = "pearson",
+  use = "complete.obs",
+  displayGrid = FALSE,
+  digits = 3,
+  title_args = list(...),
+  group_args = list(...),
+  justify_labels = "right",
+  alignPercent = 0.5,
+  title = "Corr"
+) {
+  ggally_exec_text(
+    data = data,
+    mapping = mapping,
+    alignPercent = alignPercent,
+    use = use,
+    displayGrid = displayGrid,
+    title_args = title_args,
+    group_args = group_args,
+    justify_labels = justify_labels,
+    justify_text = "left",
+    sep = if ("colour" %in% names(mapping)) ": " else ":\n",
+    title = title,
+    text_fn = function(x, y) {
+      if (is_date(x)) {
+        x <- as.numeric(x)
+      }
+      if (is_date(y)) {
+        y <- as.numeric(y)
+      }
 
-  if (! is.null(corAlignPercent)) {
-    stop("'corAlignPercent' is deprecated.  Please use argument 'alignPercent'")
-  }
-  if (! is.null(corMethod)) {
-    stop("'corMethod' is deprecated.  Please use argument 'method'")
-  }
-  if (! is.null(corUse)) {
-    stop("'corUse' is deprecated.  Please use argument 'use'")
-  }
+      corObj <- stats::cor.test(x, y, method = method, use = use)
+
+      # make sure all values have X-many decimal places
+      cor_est <- as.numeric(corObj$estimate)
+      cor_txt <- formatC(cor_est, digits = digits, format = "f")
+
+      # if stars should be added
+      if (isTRUE(stars)) {
+        cor_stars <- function(pValue) {
```

Comment:
Should be a standalone function that could be reused in other functions. A more generic version could be fount at https://github.com/larmarange/JLutils/blob/master/R/formating.R#L360